src/Components/DemographicSurvey.js

```diff
@@ -9,9 +11,18 @@ const myCss = {
 
 function DemographicSurvey({ surveyModel }) {
     const [results, setResults] = useState('');
+    const [cacheScreen, setCacheScreen] = useLocalStorage('screenerSurvey', null);
+    const [cacheDem, setCacheDem] = useLocalStorage('demographicSurvey', null);
 
-    function handleComplete(survey) {
-        setResults(survey.data);
+    async function handleComplete(survey) {
+        survey.data = { ...survey.data, uuid: cacheScreen.uuid || null };
+        let res = await putSurveyResult('demographic', survey.data);
+        setCacheDem({ ...cacheDem, uuid: cacheScreen.uuid });
+        setResults(survey.data); //is this necessary anymore?
```

Comment:
Hm yeah we definitely don't need `setResults` here, nor the useState declaration on line 13 at all

Reply:
Seems like Results is used to determine if the Demographic survey should be displayed or not. On that topic, should the Demographic survey also be reset when the Screener survey is ? Otherwise the Demographic survey can only be done once.

Reply:
There's a `result` that's created and used in the `ScreenerSurvey`, but this is a separate piece of state that is created on line 13 and not doing anything for this component. The `result` in `ScreenerSurvey` determines whether to show `Outcome`, which contains `DemographicSurvey`.

I'm pretty sure the Dem survey is reset automatically when the Screener is, because it's removed from the DOM and then reinserted once `result` (the one in `ScreenerSurvey` has a value. Let me know if that doesn't make sense, happy to discuss in Discord.

Reply:
I see what you mean about Dem survey - you're right it should be reset

Reply:
^ That can be solved by moving `const demoSurveyModel = new Survey.Model(DEMOGRAPHIC_SURVEY_MODEL);` inside the `ScreenerOutcome` component, right now it's outside at the top of the file